### DISCUSSION TITLE
work with `guideSet` object in python

### DISCUSSION THREAD

**Discussion by abearab** [BiocPy/BiocPy#5] [Ideas]
Hi, I'm a big fan of [crisprVerse](https://github.com/crisprVerse) and I found the `GuideSet` object very useful in my projects. 

> The `GuideSet` object is an extension of a `GenomicRanges` object that stores additional information about gRNAs.

However, I don't see an existing standard package to work with such data in python. I wasn't sure where to open this issue but I thought this might be a good place to start the conversation. Looking forward to hear your thoughts.

cc @Jfortin1

**Comment by jkanche** [comment]
Both @Jfortin1  and I have been thinking about this for a while. Providing the representation is probably straightforward, but supporting downstream analytical functionality might be a bit tricky. My understanding is we need some implementation or equivalent existing packages that provide similar functionality as R/Bioc's `BioStrings`: #3. 

What methods from the guideset do you more frequently use? May be thats a good starting point for us to think of how we can port that functionality.

**Comment by Jfortin1** [comment]
@abearab The first thing we are planning to do is port over the GuideSet representation, and probably most of crisprBase as well, and some of the genomics arithmetic functions in crisprDesign

**Reply by abearab** [reply]
Hey @Jfortin1 – any updates in this regard? I'm hoping to launch a project and I would love to keep everything in python!

**Comment by abearab** [comment]
Thank you @jkanche and @Jfortin1 for your quick responses!

> What methods from the guideset do you more frequently use? May be thats a good starting point for us to think of how we can port that functionality.

The way I think about this problem is this (all these are my naive understanding):
1. A user would pick a CRISPR technology, mainly the PAM sequence and spacer length are the key parameters to start.
2. Then user aims to search for potential guide sequences for a given genomics coordinates using a reference genome (potentially `GenomicRanges` object). Alternatively, a user may already have a list of guide sequences (for instance known [pooled-library](https://www.addgene.org/pooled-library/)) so it's helpful to enable loading that into a `guideset` object for downstream tasks.
3. As @Jfortin1 mentioned, a tool like `crisprDesign` would help to calculate on on-target / off-target scores. I agree this might need fresh implementations.
4. For combinatorial platforms, guide pairs are also something I can imagine a clear use case on my end.
